rl4co/models/zoo/matnet/policy.py

```diff
@@ -26,36 +41,171 @@ class MatNetPolicy(AutoregressivePolicy):
 
     def __init__(
         self,
-        env_name: str,
+        env: RL4COEnvBase,
         embedding_dim: int = 256,
         num_encoder_layers: int = 5,
         num_heads: int = 16,
         normalization: str = "instance",
         init_embedding_kwargs: dict = {"mode": "RandomOneHot"},
         use_graph_context: bool = False,
+        bias: bool = False,
         **kwargs,
     ):
-        if env_name not in ["atsp"]:
-            log.error(f"env_name {env_name} is not originally implemented in MatNet")
+        if env.name not in ["atsp", "ffsp"]:
+            log.error(f"env_name {env.name} is not originally implemented in MatNet")
+
+        if env.name == "ffsp":
+            decoder = MatNetFFSPDecoder(
+                env=env,
```

Comment:
Is passing the full `env` necessary here? It seems that in the `FFSPDecoder` no env is needed

Reply:
you are right! I will remove this line